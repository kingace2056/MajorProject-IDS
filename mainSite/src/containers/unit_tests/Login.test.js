import React from "react";
import ReactDOM from "react-dom";
import { createStore, compose, applyMiddleware, combineReducers } from "redux";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import authReducer from "../../store/reducers/auth";
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from "react-router-dom";

import Login from "../Login"

const composeEnhances = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const rootReducer = combineReducers({
    auth: authReducer
});

const store = createStore(rootReducer, composeEnhances(applyMiddleware(thunk)));

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Router><Provider store={store}><Login /></Provider></Router>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
    const tree = renderer
        .create(<Router><Provider store={store}><Login /></Provider></Router>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});