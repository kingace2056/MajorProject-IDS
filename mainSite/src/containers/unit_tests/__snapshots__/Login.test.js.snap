// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui center aligned middle aligned grid"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div
    className="column"
    style={
      Object {
        "maxWidth": 450,
      }
    }
  >
    <h2
      className="ui teal center aligned header"
    >
      Log-in to your account
    </h2>
    <span
      style={
        Object {
          "color": "red",
        }
      }
    />
    <form
      className="ui large form"
      onSubmit={[Function]}
    >
      <div
        className="ui stacked segment"
      >
        <div
          className="field"
        >
          <div
            className="ui fluid left icon input"
          >
            <input
              name="username"
              onChange={[Function]}
              placeholder="Username"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="user icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui fluid left icon input"
          >
            <input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <i
              aria-hidden="true"
              className="lock icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <button
          className="ui teal large fluid button"
          id="login_button"
          onClick={[Function]}
        >
          Login
        </button>
      </div>
    </form>
    <div
      className="ui message"
    >
      New to us? 
      <a
        aria-current={null}
        href="/signup"
        onClick={[Function]}
      >
        Sign Up
      </a>
    </div>
  </div>
</div>
`;
